aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy

```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
     void generate() {
         def sourceGenerator = newGenerator()
         sourceGenerator.generate(context)
-        def sources = context.getGeneratedJavaFiles().collectEntries([:]) {
+        def javaFiles = context.getGeneratedJavaFiles()
+        def sources = new HashMap<JavaFile, String>()
+        javaFiles.forEach {
             def writer = new StringWriter()
             it.writeTo(writer)
-            [it, writer.toString()]
+            sources.put(it, writer.toString())
```

Comment:
question: Why did you change this?

Reply:
The problem with invalid generics. The resulting object, for some reason `Map<Object, Object>`. Because of this, an error occurs when trying to compile, because it requires `Map<JavaFile, String>`

Maybe this is specific of jdk or groovy for windows

Reply:
What do you use to compile? The Gradle build just passes fine, so this hints at an IDEA bug

Reply:
No, this is not a problem of IDEA. Look to screenshots. First - run from IDEA, second - from powershell:
![изображение](https://user-images.githubusercontent.com/48474279/221610419-48217db2-83fc-4747-beea-0466cdfa5430.png)
![изображение](https://user-images.githubusercontent.com/48474279/221610511-12cb4e7e-e4ca-4da0-978b-328a2a6edb00.png)

Reply:
@melix hah, really! Reverted groovy version and now all works fine. Reverted these changes